hc/api/models.py

```diff
@@ -85,12 +86,26 @@ def get_status(self):
             return self.status
 
         now = timezone.now()
+        next_ping = self.last_ping + self.timeout
+
+        if self.status == "too often":
+            if now > next_ping:
+                return "down"
+            return self.status
 
         if self.last_ping + self.timeout + self.grace > now:
             return "up"
-
         return "down"
 
+    def is_running_too_often(self):
+        now = timezone.now()
+        if not self.last_ping:
+            return False
+
+        now = timezone.now()
```

Comment:
Great thought here. I like everything you have done in logic and execution. It looks good to me. The only thing you could work on is to remove the unused line of code. This (now = timezone.now()) is a repetition just one is enough in this function

Reply:
@Michyjones  Thanks for the feedback and noted working on the repetition.